Our solrconfig.xml relies on <lib>, but those are ignored in Solr 9.8

### Steps to recreate

1. From the blacklight repo, run `SOLR_VERSION=9.8.0 docker compose up solr`
2. Note that you get the following errors:

```
SolrCore failed to load on startup => org.apache.solr.common.SolrException: Unable to create core [blacklight-core]
[...]
Caused by: org.apache.solr.common.SolrException: Could not load conf for core blacklight-core: Error loading parsing schema
[...]
Caused by: org.apache.solr.common.SolrException: Can't load schema /var/solr/data/blacklight-core/schema.xml: Plugin init failure for [schema.xml] fieldType "text": Plugin init failure for [schema.xml] analyzer/tokenizer:  Error loading class 'solr.ICUTokenizerFactory'
[...]
Caused by: java.lang.ClassNotFoundException: solr.ICUTokenizerFactory
```

If you go to localhost:8983/solr/#/~logging, you will also see:

```
Configset references one or more <lib/> directives,&#8203; but <lib/> usage is disabled on this Solr node. Either remove all <lib/> tags from the relevant configset,&#8203; or enable use of this feature by setting 'solr.config.lib.enabled=true'
```

### Solr changes

Solr 9.8.0 includes [this patch](https://issues.apache.org/jira/browse/SOLR-16781), which recommends adding the necessary JARs to Solr's classpath.


### Impact

This is causing CI to fail.

--- a/compose.yaml
+++ b/compose.yaml
@@ -23,6 +23,7 @@ services:
     environment:
       - SOLR_PORT # Set via environment variable or use default defined in .env file
       - SOLR_VERSION # Set via environment variable or use default defined in .env file
+      - SOLR_MODULES=analysis-extras
     image: "solr:${SOLR_VERSION}"
     volumes:
       - $PWD/lib/generators/blacklight/templates/solr/conf:/opt/solr/conf

--- a/lib/generators/blacklight/templates/solr/conf/solrconfig.xml
+++ b/lib/generators/blacklight/templates/solr/conf/solrconfig.xml
@@ -16,7 +16,7 @@
     </updateLog>
   </updateHandler>
 
-  <!-- solr lib dirs -->
+  <!-- solr lib dirs, which are needed for Solr 8 compatibility but ignored in solr 9.8 and above -->
   <lib dir="${solr.install.dir:../../../..}/modules/analysis-extras/lib" />
   <lib dir="${solr.install.dir:../../../..}/contrib/analysis-extras/lib" />
   <lib dir="${solr.install.dir:../../../..}/contrib/analysis-extras/lucene-libs" />

